### PR TITLE
Correct asymmetric background map generation

### PR DESCRIPTION
**Pull request by watsonjj**
Previously, the background map returned from `make_map_background_irf` assumed a symmetric background IRF in the FoV. This PR correctly calculates the FoV coordinates to evaluate the background at. This is achieved with the `FixedPointingInfo` class, which eases the conversion from sky coordinates to FoV coordinates.

The `FixedPointingInfo` is generated from the pointing metadata contained in the `EVENTS` hdu. 

Adresses #1987

### CODE REVIEW

File: gammapy/data/pointing.py [hunk 1]
```diff
@@ -9,7 +9,102 @@
...
+        (`~astropy.coordinates.SkyCoord`)"""
+        print(self.meta.keys())
+        lon = self.meta["RA_PNT"]
+        lat = self.meta["DEC_PNT"]
+        return SkyCoord(lon, lat, unit="deg", frame="icrs")
+
+    @lazyproperty
+    def altaz_frame(self):
+        """ALT / AZ frame (`~astropy.coordinates.AltAz`)."""
+        return AltAz(obstime=self.time_start, location=self.location)
```
Comment (registerrier) [inline gammapy/data/pointing.py]: Using the middle of the time interval might give better results on average, no?

File: gammapy/data/pointing.py [hunk 2]
```diff
@@ -9,7 +9,102 @@
...
+
+        The time difference between the TSTART and TSTOP.
+        """
+        return self.time_stop - self.time_start
+
+    @lazyproperty
+    def radec(self):
+        """RA / DEC pointing postion from table
+        (`~astropy.coordinates.SkyCoord`)"""
+        print(self.meta.keys())
```
Comment (registerrier) [inline gammapy/data/pointing.py]: Why the `print` here?

File: gammapy/cube/tests/test_background.py [hunk 3]
```diff
@@ -51,14 +122,82 @@ def geom(map_type, ebounds):
...
+            skydir=fixed_pointing_info.radec
+        ),
     )
 
     assert m.data.shape == pars["shape"]
     assert m.unit == ""
     assert_allclose(m.data.sum(), pars["sum"], rtol=1e-5)
+
+
+def test_make_map_background_irf_constant(bkg_3d_constant, fixed_pointing_info):
```
Comment (registerrier) [inline gammapy/cube/tests/test_background.py]: Maybe the tests here could be parametrized to limit the code duplication?

File: gammapy/cube/tests/test_background.py [hunk 4]
```diff
@@ -18,10 +27,70 @@ def bkg_3d():
     return Background3D.read(filename, hdu="BACKGROUND")
 
 
-def geom(map_type, ebounds):
+@pytest.fixture(scope="session")
+def bkg_3d_constant():
```
Comment (registerrier) [inline gammapy/cube/tests/test_background.py]: Maybe reduce code duplication by passing argument to `bkg_3d`?

File: gammapy/data/pointing.py (hunk 2 above)
Comment (watsonjj) [reply gammapy/data/pointing.py]: Removed

File: gammapy/data/pointing.py (hunk 1 above)
Comment (watsonjj) [reply gammapy/data/pointing.py]: Okay, done

**Comment by watsonjj** [comment]
I've tried to remove code duplication, and I have also improved the testing for different symmetries. Following #2035 the tests should pass.

**Comment by adonath** [comment]
The Travis fails seem unrelated. I'll merge and fix remaining test fails in master. Thanks @watsonjj!

**Pull request merged**